engine/src/main/java/org/entando/entando/aps/system/services/api/cache/ApiCatalogManagerCacheWrapper.java

```diff
@@ -0,0 +1,158 @@
+
+import org.apache.commons.lang3.StringUtils;
+import org.entando.entando.aps.system.exception.CacheItemNotFoundException;
+import org.entando.entando.aps.system.services.api.IApiCatalogDAO;
+import org.entando.entando.aps.system.services.api.model.ApiMethod;
+import org.entando.entando.aps.system.services.api.model.ApiResource;
+import org.entando.entando.aps.system.services.api.model.ApiService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.Cache;
+
+import com.agiletec.aps.system.common.AbstractCacheWrapper;
+import com.agiletec.aps.system.exception.ApsSystemException;
+
+public class ApiCatalogManagerCacheWrapper extends AbstractCacheWrapper implements IApiCatalogManagerCacheWrapper {
+
+	private final Logger logger = LoggerFactory.getLogger(getClass());
+
+	@Override
+	protected String getCacheName() {
+		return APICATALOG_MANAGER_CACHE_NAME;
+	}
+
+	@Override
+	public void initCache(Map<String, ApiResource> resources, IApiCatalogDAO apiCatalogDAO) throws ApsSystemException {
+		try {
+			Cache cache = this.getCache();
+			this.releaseCachedObjects(cache);
+			apiCatalogDAO.loadApiStatus(resources);
+			this.insertObjectsOnCache(cache, resources, apiCatalogDAO);
+		} catch (Throwable t) {
+			logger.error("Error bootstrapping ApiCatalog cache", t);
+			throw new ApsSystemException("Error bootstrapping ApiCatalog cache", t);
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public Map<String, ApiResource> getMasterResources() {
+		return this.get(this.getCache(), APICATALOG_RESOURCES_CACHE_NAME, Map.class);
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public Map<String, ApiService> getMasterServices() {
+		return this.get(this.getCache(), APICATALOG_SERVICES_CACHE_NAME, Map.class);
+	}
+
+	@Override
+	public void addService(ApiService apiService) {
+		this.manage(apiService.getKey(), apiService, Action.ADD);
+	}
+
+	@Override
+	public void updateService(ApiService apiService) {
+		this.manage(apiService.getKey(), apiService, Action.UPDATE);
+	}
+
+	@Override
+	public void removeService(String key) {
+		this.manage(key, null, Action.DELETE);
+	}
+
+	protected void insertObjectsOnCache(Cache cache, Map<String, ApiResource> resources, IApiCatalogDAO apiCatalogDAO) throws ApsSystemException {
+		Map<String, ApiService> services = new HashMap<>();
+		try {
+			List<ApiMethod> apiGETMethods = buildApiGetMethods(resources);
+			services = apiCatalogDAO.loadServices(apiGETMethods);
+
+			for (Map.Entry<String, ApiService> entry : services.entrySet()) {
+				cache.put(APICATALOG_SERVICE_CACHE_NAME_PREFIX + entry.getKey(), entry.getValue());
+			}
+
+		} catch (Throwable t) {
+			services = new HashMap<String, ApiService>();
+			logger.error("Error loading Services definitions", t);
+			throw new ApsSystemException("Error loading Services definitions", t);
+		}
+		cache.put(APICATALOG_RESOURCES_CACHE_NAME, resources);
+		cache.put(APICATALOG_SERVICES_CACHE_NAME, services);
```

Comment:
Is this for sure where these belong with respect to the try call? If an exception is thrown partway through the loop we would have a mix of in and out of cache objects from the services list. May not matter and sort of a fringe case but these two calls at the end maybe out of place